File: utils/watermarkremover.py
import PyPDF2

def add_watermark(input_pdf_path, output_pdf_path, watermark_text):
    with open(input_pdf_path, 'rb') as input_file:
        pdf_reader = PyPDF2.PdfReader(input_file)
        pdf_writer = PyPDF2.PdfWriter()

        for page_num in range(len(pdf_reader.pages)):
            page = pdf_reader.pages[page_num]
            page.merge_page(watermark_page(page.mediabox.width, page.mediabox.height, watermark_text))

            pdf_writer.add_page(page)

        with open(output_pdf_path, 'wb') as output_file:
            pdf_writer.write(output_file)

def watermark_page(width, height, text):
    watermark = PyPDF2.pdf.PageObject.createBlankPage(width, height)
    watermark.mergeScaledTranslatedPage(watermark_page_content(text), 0.5, 0.5)
    return watermark

def watermark_page_content(text):
    c = PyPDF2.pdf.ContentStream([
        PyPDF2.pdf.PdfTextObject(text),
    ])
    return PyPDF2.pdf.PageObject.createContentStream(c)

# Usage
input_pdf = 'frdA181123A190070_FirstRanker.com.pdf'
output_pdf = 'output.pdf'
watermark_text = 'FirstRanker'

add_watermark(input_pdf, output_pdf, watermark_text)

File: utils/watermarkremover.js
const { PDFDocument, rgb, StandardFonts } = require('@pdf-lib/core');
const fs = require('fs');

// Function to remove watermark
async function removeWatermark(inputPdfBytes, outputPdfPath, watermarkText) {
    const pdfDoc = await PDFDocument.load(inputPdfBytes);

    // Iterate through each page and remove the watermark text
    pdfDoc.getPages().forEach(page => {
        const content = page.getTextContent();
        if (content.some(line => line.str.includes(watermarkText))) {
            const watermarkTextIndex = content.findIndex(line => line.str.includes(watermarkText));
            page.setTextContent(content.filter((_, index) => index !== watermarkTextIndex));
        }
    });

    // Save the modified PDF to a new file
    const modifiedPdfBytes = await pdfDoc.save();
    fs.writeFileSync(outputPdfPath, modifiedPdfBytes);
}

// Function to add watermark
async function addWatermark(inputPdfBytes, outputPdfPath, watermarkText) {
    const pdfDoc = await PDFDocument.load(inputPdfBytes);
    const font = await pdfDoc.embedFont(StandardFonts.Helvetica);

    // Iterate through each page and add the watermark text
    pdfDoc.getPages().forEach(page => {
        const { width, height } = page.getSize();
        const textSize = 30; // Adjust the text size as needed
        const textWidth = font.widthOfTextAtSize(watermarkText, textSize);
        const textHeight = font.heightAtSize(textSize);
        const textX = (width - textWidth) / 2;
        const textY = (height - textHeight) / 2;

        page.drawText(watermarkText, {
            x: textX,
            y: textY,
            size: textSize,
            font: font,
            color: rgb(0.5, 0.5, 0.5), // Adjust the color as needed
        });
    });

    // Save the modified PDF to a new file
    const modifiedPdfBytes = await pdfDoc.save();
    fs.writeFileSync(outputPdfPath, modifiedPdfBytes);
}

// Usage
const inputPdfPath = 'input.pdf';
const outputPdfPath = 'output.pdf';
const watermarkTextToRemove = 'Watermark Text to Remove';
const watermarkTextToAdd = 'Your Watermark Text';

const inputPdfBytes = fs.readFileSync(inputPdfPath);

// Remove watermark
removeWatermark(inputPdfBytes, outputPdfPath, watermarkTextToRemove)
    .then(() => console.log('Watermark removed successfully'))
    .catch(error => console.error('Error removing watermark:', error));

// Add watermark
addWatermark(inputPdfBytes, outputPdfPath, watermarkTextToAdd)
    .then(() => console.log('Watermark added successfully'))
    .catch(error => console.error('Error adding watermark:', error));
